src/rulesets/Base/indexing.jl

```diff
@@ -125,6 +179,16 @@ function frule((_, ẋ, v̇), ::typeof(setindex!), x::AbstractArray, v, inds...)
     return setindex!(x, v, inds...), setindex!(ẋ, v̇, inds...)
 end
 
+#####
+##### unsafe_getindex
+#####
+
+# This is called by e.g. `iterate(1:0.1:2)`,
+# and fixes https://github.com/FluxML/Zygote.jl/issues/1247
+
+function rrule(cfg::RuleConfig{>:HasReverseMode}, ::typeof(Base.unsafe_getindex), x::AbstractRange, i::Integer)
+  return rrule_via_ad(cfg, getindex, x, i)
+end
```

Comment:
Not strictly related, but also in need of fixing. A bit unclear how to test this.

Reply:
I believe we now can test these since
https://github.com/JuliaDiff/ChainRulesTestUtils.jl/pull/243

Reply:
Ah that's not the problem, it's the restriction to AbstractRange.

Reply:
Fixed by widening the rule